src/main/java/net/dv8tion/jda/core/entities/Message.java

```diff
@@ -1254,4 +1263,99 @@ public boolean isImage()
             return height > 0 && width > 0;
         }
     }
+
+    /**
+     * Represents a {@link net.dv8tion.jda.core.entities.Message message} activity like Spotify
+     */
+    class Activity
+    {
+        private final ActivityType type;
+        private final String party_id;
+
+        Activity(ActivityType type, String party_id)
+        {
+            this.type = type;
+            this.party_id = party_id;
+        }
+
+        /**
+         *
+         * @return the type of the activity
+         */
+        public ActivityType getType()
+        {
+            return type;
+        }
+
+        /**
+         *
+         * @return {@link java.lang.String string} containing the activity service and its host (a {@link net.dv8tion.jda.core.entities.User} id)<br>
+         *         For example : "spotify:86699011792191488"
+         */
+        public String getPartyId()
+        {
+            return party_id;
+        }
+
+        /**
+         * Shortcut for grabbing the service of the activity
+         *
+         * @return the service name like "spotify"
+         */
+        public String getService()
+        {
+            return party_id.split(":", 2)[0];
+        }
+
+        /**
+         * This is a shortcut for getting the host id of the activity
+         *
+         * @return the service name like "86699011792191488"
+         */
+        public String getHostId()
+        {
+            return party_id.split(":", 2)[1];
+        }
+
+        /* JDA internal */
+        static Activity parseJSON(JSONObject jsonObject)
+        {
+            final ActivityType type = ActivityType.fromId(jsonObject.getInt("type"));
+            if (type == ActivityType.PARTY)
+            {
+                return new Activity(type, jsonObject.getString("party_id"));
+            }
+            WebSocketClient.LOG.debug("Received an unknown activity type in a message. JSON: {}", jsonObject);
+            return new Activity(type, "");
+        }
+    }
+
+    // missing data for more
+    enum ActivityType
+    {
+        PARTY(3),
+        UNKNOWN(-1);
```

Comment:
I'd like to see research into other types

Reply:
provide more data for it i mean sure there are types but from no data i cant generate stuff